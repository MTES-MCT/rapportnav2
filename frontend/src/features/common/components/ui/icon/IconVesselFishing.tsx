import Icon from '../../../../../assets/images/navire_peche_pro.svg'

function IconVesselFishing() {
  return <img src={Icon} alt="Navire de pÃªche professionnelle" height={32} />
}

export default IconVesselFishing
