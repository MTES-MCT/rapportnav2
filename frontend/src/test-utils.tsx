import React, { ReactElement } from 'react'
import { render, renderHook, RenderOptions } from '@testing-library/react'
import UIThemeWrapper from './ui/ui-theme-wrapper'
import { BrowserRouter } from 'react-router-dom'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <UIThemeWrapper>
      <BrowserRouter>{children}</BrowserRouter>
    </UIThemeWrapper>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

const customRenderHook = (hook: any, options?: Omit<RenderOptions, 'wrapper'>) =>
  renderHook(() => hook(), { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
export { customRenderHook as renderHook }
