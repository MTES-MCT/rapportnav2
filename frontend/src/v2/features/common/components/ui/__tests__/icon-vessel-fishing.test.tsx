import { render } from '../../../../../../test-utils'
import IconVesselFishing from '../icon-vessel-fishing'

describe('IconVesselFishing', () => {
  it('should match the snapshot', () => {
    const wrapper = render(<IconVesselFishing />)
    expect(wrapper).toMatchSnapshot()
  })
})
