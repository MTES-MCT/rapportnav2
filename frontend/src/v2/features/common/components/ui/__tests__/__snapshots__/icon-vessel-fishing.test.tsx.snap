// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconVesselFishing > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="Navire de pêche pro"
        src="data:image/svg+xml,%3csvg%20height='32'%20viewBox='0%200%2051%2032'%20width='51'%20xmlns='http://www.w3.org/2000/svg'%3e%3cclipPath%20id='a'%3e%3cpath%20d='m0%200h51v32h-51z'/%3e%3c/clipPath%3e%3cg%20clip-path='url(%23a)'%3e%3cpath%20d='m0%200h51v32h-51z'%20fill='transparent'/%3e%3cg%20fill='%233b4559'%3e%3cpath%20d='m152.766%20120.266h28.018v4.519h-28.018zm0%200'%20transform='translate(-135.519%20-107.075)'/%3e%3cpath%20d='m51.766%2061.266h9.829v3.05h-9.829zm0%200'%20transform='translate(-45.917%20-54.551)'/%3e%3cpath%20d='m51.766%20102.266h9.829v6.553h-9.829zm0%200'%20transform='translate(-45.917%20-91.078)'/%3e%3cpath%20d='m5.066%20174.266h-5.066l1.438%209.716h43.038l6.524-9.716zm0%200'%20transform='translate(0%20-155.371)'/%3e%3cpath%20d='m81.766.266h2.934v5.31h-2.934zm0%200'%20transform='translate(-72.53%20-.266)'/%3e%3cpath%20d='m15.07%20276.074h40.235l1.237-1.808h-41.742zm0%200'%20transform='translate(-13.125%20-244.074)'/%3e%3c/g%3e%3c/g%3e%3c/svg%3e"
      />
    </div>
    <div
      class="rs-toast-provider"
    >
      <div
        class="rs-toast-container rs-toast-container-top-center"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-center"
      />
      <div
        class="rs-toast-container rs-toast-container-top-start"
      />
      <div
        class="rs-toast-container rs-toast-container-top-end"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-start"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-end"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="Navire de pêche pro"
      src="data:image/svg+xml,%3csvg%20height='32'%20viewBox='0%200%2051%2032'%20width='51'%20xmlns='http://www.w3.org/2000/svg'%3e%3cclipPath%20id='a'%3e%3cpath%20d='m0%200h51v32h-51z'/%3e%3c/clipPath%3e%3cg%20clip-path='url(%23a)'%3e%3cpath%20d='m0%200h51v32h-51z'%20fill='transparent'/%3e%3cg%20fill='%233b4559'%3e%3cpath%20d='m152.766%20120.266h28.018v4.519h-28.018zm0%200'%20transform='translate(-135.519%20-107.075)'/%3e%3cpath%20d='m51.766%2061.266h9.829v3.05h-9.829zm0%200'%20transform='translate(-45.917%20-54.551)'/%3e%3cpath%20d='m51.766%20102.266h9.829v6.553h-9.829zm0%200'%20transform='translate(-45.917%20-91.078)'/%3e%3cpath%20d='m5.066%20174.266h-5.066l1.438%209.716h43.038l6.524-9.716zm0%200'%20transform='translate(0%20-155.371)'/%3e%3cpath%20d='m81.766.266h2.934v5.31h-2.934zm0%200'%20transform='translate(-72.53%20-.266)'/%3e%3cpath%20d='m15.07%20276.074h40.235l1.237-1.808h-41.742zm0%200'%20transform='translate(-13.125%20-244.074)'/%3e%3c/g%3e%3c/g%3e%3c/svg%3e"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
